src/test/java/HomePageTests.java

```diff
@@ -0,0 +1,71 @@
+import Pages.HomePage;
+import Pages.LoginPage;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Keys;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+
+public class HomePageTests extends BaseTest{
+    String newPlaylistName = "Updated playlist";
+    @Test
+    public void renamePlaylist() throws InterruptedException {
```

Comment:
Please, delete `throws InterruptedException`

Thread. sleep is bad! It blocks the current thread and renders it unusable for further work.
Using Thread. sleep() frequently in an automation framework is not a good practice.

Please use the Explicit Wait command, the WebDriver is directed to wait until a certain condition occurs before proceeding with executing the code.

Reply:
I tried to do the explicit wait command with the webdriver but it did not work. I had Czar look at my code and he couldn't get it to work either and said there is probably an issue with the koel website which is why we included the thread.sleep, see comment from Czar attached.
Thanks!
![czar comment](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/137116665/de2b85e8-826c-4358-a2a7-d41e42eabfcd)

Reply:
**Step 1**

public class HomePage extends BasePage

```
    public HomePage changePlaylistInputField(String name) {
        findElement(playListInputField).sendKeys(Keys.chord(Keys.CONTROL, "a", Keys.BACK_SPACE));
        findElement(playListInputField).sendKeys(name);
        findElement(playListInputField).sendKeys(Keys.ENTER);
        return this;
    }
```
![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/a703672c-1778-4a58-92c4-bbb39437dc4a)

**The method then performs the following actions:**

- Selects the playListInputField element on the page.
- Sends a series of keystrokes to the input field, which effectively clears it by first selecting all text (Keys.chord(Keys.CONTROL, "a")), then deleting it (Keys.BACK_SPACE).
- Sends the contents of the "name" argument to the input field.
- Sends the "ENTER" keystroke to submit the form.
- Returns an instance of the "HomePage" class.


**Step 2**

public class HomePageTests extends BaseTest

```
    @Test
    public void renamePlaylist() {
        // GIVEN
        String newPlaylistName = "Updated playlist";
        LoginPage loginPage = new LoginPage(driver);
        HomePage homePage = new HomePage(driver);

        // WHEN
        loginPage.login();
        homePage.doubleClickPlaylist().changePlaylistInputField(newPlaylistName);

        // THEN
        Assert.assertTrue(homePage.doesPlaylistExist().contains(newPlaylistName));

    }
```
![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/da43db72-cbd6-48fe-9638-9375361cb802)

**The method uses the following steps:**

- Sets the new playlist name to "Updated playlist".
- Creates instances of the "LoginPage" and "HomePage" classes using the "driver" object.
- Calls the "login" method of the "LoginPage" instance to log in to the web application.
- Calls the "doubleClickPlaylist" method of the "HomePage" instance to select a playlist.
- Calls the "changePlaylistInputField" method of the "HomePage" instance to rename the playlist with the new name.
- Calls the "doesPlaylistExist" method of the "HomePage" instance to check if the playlist with the new name exists on the page.
- Uses the "Assert" class to verify that the new playlist name is indeed present on the page.


**RESULT**

![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/a54530d8-8a7d-4464-aae5-38c7beec6fe4)


**HINT**

For example

public class BasePage

```
    protected void doubleClick(WebElement webElement) {
        wait.until(ExpectedConditions.elementToBeClickable(webElement));
        actions.doubleClick(webElement).perform();
    }
```
![image](https://github.com/Test-Pro-Bootcamp/A47-QA-Automation-Framework/assets/104761482/bc68da1e-d616-4a94-b5ff-2ab6a516bab9)

**The method performs the following actions:**

- Waits for the "webElement" to become clickable using the ExpectedConditions.elementToBeClickable method.
- Initializes an instance of the Actions class and assigns it to the "actions" variable.
- Sends a double-click event to the "webElement" to simulate a double-click event using the actions.doubleClick(webElement) method.
- Executes the sequence of actions using the perform() method.